### PR TITLE
#1201 Hotfix twilio delivery status payload

### PR DESCRIPTION
**Pull request by jakehova**
# Description

#1201 

The Delivery Status handler was not including the original delivery status payload in the service callback request.  

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Created service callback with "include_provider_payload" = true and verified that provider payload came through with service callback
- Updated service callback with "include_provider_payload" = false and verified that provider payload DID NOT came through with service callback

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by k-macmillan — APPROVED** [review]

File: app/celery/service_callback_tasks.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def create_delivery_status_callback_data(notification, service_callback_api, pro
     }
 
     # add the property 'provider_payload when provider payload is not an empty dictionary
-    if not provider_payload:
+    if provider_payload:
         data['provider_payload'] = provider_payload
```
Comment (kalbfled) [inline app/celery/service_callback_tasks.py:233]: I recommend deleting lines 209-213 and changing these lines to:
```
data["provider_payload"] = {} if provider_payload is None else provider_payload
```

**Review by kalbfled — CHANGES_REQUESTED** [review]

**Review by cris-oddball — CHANGES_REQUESTED** [review]
Please open a ticket in this sprint and include that ticket number in the description.

File: app/celery/service_callback_tasks.py [hunk 2]
```diff
@@ -109,7 +114,8 @@ def send_complaint_to_service(self, service_callback_id, complaint_data):
         except self.MaxRetriesExceededError:
             current_app.logger.error(
                 f"Retry: {self.name} has retried the max num of times for {logging_tags}, url "
-                f"{service_callback.url}. exc: {e}")
+                f"{service_callback.url}. exc: {e}"
+            )
```
Comment (cris-oddball) [inline app/celery/service_callback_tasks.py:118]: These should all be %s for logging, not f-strings, yes?

File: app/celery/service_callback_tasks.py [hunk 3]
```diff
@@ -119,49 +125,59 @@ def send_complaint_to_service(self, service_callback_id, complaint_data):
...
         )
         current_app.logger.info(
-            f'Successfully sent complaint email to va-notify. notification_id: {complaint.notification_id}'
+            f"Successfully sent complaint email to va-notify. notification_id: {complaint.notification_id}"
         )
 
     except Exception as e:
         current_app.logger.exception(
-            f'Problem sending complaint to va-notify for notification {complaint.notification_id}: {e}'
+            f"Problem sending complaint to va-notify for notification {complaint.notification_id}: {e}"
```
Comment (cris-oddball) [inline app/celery/service_callback_tasks.py:158]: logging should use %s, not f strings?

File: app/celery/service_callback_tasks.py [hunk 4]
```diff
@@ -192,7 +205,8 @@ def send_inbound_sms_to_service(self, inbound_sms_id, service_id):
         except self.MaxRetriesExceededError:
             current_app.logger.error(
                 f"Retry: {self.name} has retried the max num of times for {logging_tags}, url "
-                f"{service_callback.url}. exc: {e}")
+                f"{service_callback.url}. exc: {e}"
+            )
```
Comment (cris-oddball) [inline app/celery/service_callback_tasks.py:209]: %s, rather than f-strings?

**Review by cris-oddball — CHANGES_REQUESTED** [review]

**Review by kalbfled — APPROVED** [review]

File: app/celery/service_callback_tasks.py (hunk 1 above)
Comment (kalbfled) [reply app/celery/service_callback_tasks.py:257]: I made this change.

**Review by k-macmillan — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**